src/Module/SourceRetriever/AddonsUrlSourceRetriever.php

```diff
@@ -295,6 +297,21 @@ private function computeAuthentication(string $source): array
         ];
     }
 
+    private function addCustomHeaderIfNeeded(array $options): array
+    {
+        if (!is_array($options['headers'])) {
+            $options['headers'] = [];
+        }
+        $customHeaderKey = getenv('ADDONS_API_HEADER_KEY');
```

Comment:
Seems to be a repetition

WDYT about this method just getting the customHeaders and returning an array $a[$customHeaderKey] = $customHeaderValue, and then you can deal with array_merge where you need it